src/Abstractions/NexusMods.Abstractions.Settings/Types/ConfigurablePath.cs

```diff
@@ -0,0 +1,48 @@
+using JetBrains.Annotations;
+using NexusMods.Paths;
+
+namespace NexusMods.Abstractions.Settings;
+
+/// <summary>
+/// Represents a configurable path. This is made up of a <see cref="KnownPath"/>
+/// base directory and a <see cref="RelativePath"/> file part.
+/// </summary>
+[PublicAPI]
+public readonly struct ConfigurablePath
```

Comment:
> This isn't supposed to be a User configurable path from what I gather?
> It seems more of a convenience class for the tests

No? As the name suggests, this is a configurable path. I don't know what gave the impression that this is just for tests, but it isn't. Any setting that refers to a path should use this type.

Reply:
It is composed of a `KnownPath` and a `RelativePath`. `KnownPath` is an `enum`. This makes what this can point to actually pretty limited.
I we could not use this as is to ask users where to put the FileStore for example, unless this is changed later on.